components/Observer.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+  <div class="observer" />
+</template>
+
+<script>
+export default {
+  props: {
```

Comment:
The thing it is replacing has a distance parameter which is very useful, because it allows you to start fetching more content in time for it to be there when you need it.  Can we do that with this approach, e.g. by artificially positioning the element?

Reply:
More generally, I think I would want to preserve the existing interface.  It's a bad interface, but I wouldn't like to grapple both with subtle bugs in this, and bugs introduced by us changing the interface in loads of places.

Reply:
By "distance" do you mean that it starts fetching and loading when the bottom of the list is not yet displayed but is like 100px away?  There is a threshold property to the intersection observer but I think that only applies to whether it triggers at the top or bottom of the element being watched.  I'd need to investigate the best way to trigger when it's not yet quite displayed.
We can probably just create a similar component to the last one and wrap it all up in there probably using slots.  Just thinking off the top of my head but I'm sure we could make it work.